### PR TITLE
Support running apk with more than one activity

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

File: packages/flutter_tools/lib/src/application_package.dart [hunk 1]
```diff
@@ -116,9 +124,12 @@ class AndroidApk extends ApplicationPackage {
     for (xml.XmlElement category in document.findAllElements('category')) {
       if (category.getAttribute('android:name') == 'android.intent.category.LAUNCHER') {
```
Comment (mravn-google) [inline packages/flutter_tools/lib/src/application_package.dart]: According to `https://stackoverflow.com/questions/25219551/what-is-the-meaning-of-android-intent-action-main` we should be looking for `android.intent.action.MAIN` instead.

File: packages/flutter_tools/lib/src/application_package.dart [hunk 2]
```diff
@@ -349,34 +360,127 @@ class ApplicationPackageStore {
...
+    final Element application = manifest.firstElement('application');
+    assert(application != null);
+
+    final Iterable<Entry> activities = application.allElements('activity');
+
+    Element launchActivity;
+    for (Element activity in activities) {
+      final Attribute enabled = activity.firstAttribute('android:enabled');
+      if (enabled == null || enabled.value.contains('0xffffffff')) {
+        launchActivity = activity;
```
Comment (mravn-google) [inline packages/flutter_tools/lib/src/application_package.dart]: This appears to imply that we get the last launchable activity. Android Studio seems to start the first such thing.

File: packages/flutter_tools/test/application_package_test.dart [hunk 3]
```diff
@@ -18,13 +18,21 @@ import 'src/context.dart';
 
 void main() {
   group('ApkManifestData', () {
-    testUsingContext('parse sdk', () {
-      final ApkManifestData data =
-          ApkManifestData.parseFromAaptBadging(_aaptData);
+    testUsingContext('parse sdk with explicit enabled activity', () {
```
Comment (mravn-google) [inline packages/flutter_tools/test/application_package_test.dart]: [nit] Do we need the context in these tests?
[nit] I'd name these tests from the intended functionality of `ApkManifestData`, as in `selects explicitly enabled main activity`.

File: packages/flutter_tools/lib/src/application_package.dart [hunk 4]
```diff
@@ -349,34 +360,127 @@ class ApplicationPackageStore {
   }
 }
 
+class Entry {
```
Comment (mravn-google) [inline packages/flutter_tools/lib/src/application_package.dart]: `Entry` and subclasses could be private?

**Review by mravn-google — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/application_package.dart (hunk 1 above)
Comment (mravn-google) [reply packages/flutter_tools/lib/src/application_package.dart]: That was intended as a question. `android.intent.action.ASK`

**Comment by szakarias** [comment]
PTAL

File: packages/flutter_tools/lib/src/application_package.dart [hunk 5]
```diff
@@ -349,34 +360,128 @@ class ApplicationPackageStore {
...
+  }
+
+  void addChild(_Entry child) {
+    children.add(child);
+  }
+
+  _Attribute firstAttribute(String name) {
+    return children.firstWhere(
+        (_Entry e) => e is _Attribute && e.key.startsWith(name),
+        orElse: () => null);
```
Comment (mravn-google) [inline packages/flutter_tools/lib/src/application_package.dart]: Trailing commas ftw!

File: packages/flutter_tools/lib/src/application_package.dart [hunk 6]
```diff
@@ -349,34 +360,128 @@ class ApplicationPackageStore {
...
+  }
+
+  _Element firstElement(String name) {
+    return children.firstWhere(
+        (_Entry e) => e is _Element && e.name.startsWith(name),
+        orElse: () => null);
```
Comment (mravn-google) [inline packages/flutter_tools/lib/src/application_package.dart]: Trailing commas ftw`,`

File: packages/flutter_tools/lib/src/application_package.dart [hunk 7]
```diff
@@ -349,34 +360,128 @@ class ApplicationPackageStore {
...
+  }
+}
+
+class _Attribute extends _Entry {
+  String key;
+  String value;
+
+  _Attribute.fromLine(String line, _Element parent) {
+    //     A: android:label(0x01010001)="hello_world" (Raw: "hello_world")
+    final List<String> keyVal = line.substring(line.indexOf('A: ') + 3).split('=');
```
Comment (mravn-google) [inline packages/flutter_tools/lib/src/application_package.dart]: 3 is a magic constant---I suppose it is a shorthand for `'A: '.length`?
Maybe extract `'A: '` as a `const String attributePrefix = 'A: ';` and use that in both places.

File: packages/flutter_tools/lib/src/application_package.dart [hunk 8]
```diff
@@ -349,34 +360,128 @@ class ApplicationPackageStore {
...
-        for (Match m in keyValueRegex.allMatches(line)) {
-          entries[m.group(1)] = m.group(2);
+      if (level > currentElement.level) {
+        switch (trimLine[0]) {
+          case 'A':
+            currentElement
+                .addChild(new _Attribute.fromLine(line, currentElement));
+            break;
+          case 'E':
+            final _Element element = new _Element.fromLine(line,currentElement);
```
Comment (mravn-google) [inline packages/flutter_tools/lib/src/application_package.dart]: Missing space after `,`

**Review by mravn-google — APPROVED** [review]
LGTM

**Pull request merged**